Events json first draft

dlebauer opened the pull request.

dlebauer committed:
added events schema

dlebauer committed:
Merge branch 'master' into events

dlebauer committed:
formatted brapi.yaml

dlebauer committed:
added hand written events.json file

dlebauer committed:
minor cleanup of events.json

dlebauer committed:
Merge branch 'master' into events

dlebauer committed:
reverted brapi.yaml that had strange whitespace changes

dlebauer committed:
updated events.json w/ ICASA terms

dlebauer committed:
minor updates to events.json

dlebauer committed:
merge updates to events.json from events_update branch

KristinaRiemer commented on data/events.json:
```diff
@@ -1,424 +1,479 @@
   {
-    "eventDbId": ["6000000077"],
-    "eventDescription": ["Liquid UAN (nitrogen) added through drip irrigation system at rate of 28 kg/ha "],
+    "eventDbId": "6000000075",
```
This needs a studyDbId, it's probably 6000000010.

KristinaRiemer commented on data/events.json:
```diff
@@ -1,424 +1,479 @@
     ],
-    "eventType": ["tillage"]
+    "eventType": "tillage",
+    "observationUnitDbIds": [ "6000005230", "6000005231", "6000005232", "6000005233", "6000005234", "6000005235", "6000005236", "6000005237", "6000005238", "6000005239", "6000005240", "6000005241", "6000005242", "6000005243", "6000005244", "6000005245", "6000005246", "6000005247", "6000005248", "6000005249", "6000005250", "6000005251", "6000005252", "6000005253", "6000005254", "6000005255", "6000 ...
```
Both `observationUnitDbId` and `observationUnitDbIds` are used throughout, so should pick one to use in all instances.

dlebauer replied:
from Peter:  "observationUnitDbId (lower case 'o', no 's') if it is a single DbId, and you should see observationUnitDbIds (lower case 'o', with an 's') if it is an array of DbIds."

I've fixed these

KristinaRiemer reviewed (commented):
Some minor fixes

dlebauer committed:
fixed issues in events.json:
- singular observationUnitDbId
- added missing StudyDbId

dlebauer committed:
fix discreteDates array brackets and quotes

dlebauer committed:
fixed plural observationUnitDbId-->observationUnitDbIds

dlebauer reviewed (commented):
- [x] change time series to valuesByDate
- [x] change irrigation time series to char
- [x] add "parameterDescription", "codeValueDescription"

dlebauer committed:
update time series
- valueByDate field
- character type w/ quotes

dlebauer committed:
add "codeValueDescription"

dlebauer committed:
add parameterDescription

dlebauer commented:
@BrapiCoordinatorSelby could you please review this events.json file? I've included the changes discussed at the last meeting in the last three commits. But I am not confident that I got it correct ... 

- [change time series to valuesByDate](https://github.com/terraref/brapi/pull/56/commits/b78810a107fe981d7933103ca9254dc81d140980) 
- [add "codeValueDescription"](https://github.com/terraref/brapi/pull/56/commits/5a304aafe93c9fef5c217fa7a59acfd7d815d28a)
- [add "parameterDescription"](https://github.com/terraref/brapi/pull/56/commits/505eccb1f8966c739b45bee63912139c1c0f4306)

KristinaRiemer commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+        "code": "IROP",
+        "name": "irrigation_operation",
+        "value": "IR012",
+        "units": "code",
```
```suggestion
        "unit": "code",
```

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+          "name": "V-Ripper",
+          "unit": "code",
+          "codeValueDescription": "Standard V-Ripper (TI001)"
+        }
```
```
{
  "code": "tiimp",   //ICASA code for the parameter (tillage implement)
  "name": "tillage_implement",  // Name for 'tiimp'
  "parameterDescription": "Implement or tool used for tillage", // desc of 'tiimp'
  "unit": "code", // units say it is a encoded value
  "value": "TI001"  // value of the parameter
  "codeValueDescription": "Standard V-Ripper (TI001)",  // desc of the encode value
}
```

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+          "code": "IRRIG",
+          "name": "irrigation_applied",
+          "value": "y"
+        },
```
according to ICASA, this should have `"unit": "code"` and probably a `"codeValueDescription": "Yes irrigation was applied"`

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+          "parameterDescription":"Irrigation amount, depth of water",
+          "unit": "mm",
+          "valuesByDate": ["4.5", "2.6", "0.4", "27.9", "14", "0", "14", "0", "14", "0", "14", "0", "14", "0", "2.7", "2.7", "14"]
+        }
```
I'm not sure what happened here, you have 14 dates and 17 values

dlebauer replied:
wow, thanks for counting! Not sure how I got it wrong, but was missing the first three days. Original data in https://github.com/cct-datascience/ardn-terra-ref/blob/b0b8c1b3cc21233d1d65710273f86b54c34f53de/irrigation/season_4_irrigation.csv

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+          "code": "CH_TARGETS",
+          "name": "chemical_applic_target",
+          "value": "grub"
+        }
```
needs `unit` 
ICASA says its a code, but I couldnt find the list of encoded values. so maybe just `"unit": "text"`

dlebauer replied:
there is a list in the Pest_codes worksheet. It didn't include 'grub' but does include 'Cutworm' which is one of the targets ... 

@BrapiCoordinatorSelby are parameterDescription and codeValueDescription required? Did I get the use correct?

BrapiCoordinatorSelby replied:
I think `parameterDescription` is optional, it is just extra human readable text to explain the parameter. Depending on the use case, parameter, and data available, it may or may not be included.

`codeValueDescription` is also optional but with some restrictions. The intent of `codeValueDescription` is to add extra human readable information to a value that is an ICASA code. 
If `unit` does not equal `"code"`, then `codeValueDescription` should not be included. I believe in your update, there were a few places that broke this rule.
If `unit` does equal `"code"`, then I would strongly suggest including `codeValueDescription`, though not required.

dlebauer replied:
thanks!

KristinaRiemer replied:
Linking to the [spec](https://brapinewconceptpreview.docs.apiary.io/#/reference/agronomy-and-crop-modeling/get-events) to remind myself where these rules are located.

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+          "code": "TI012",
+          "name": "Cultivator, row"
+        }
+    ],
```
```
{
  "code": "tiimp",
  "name": "tillage_implement",
  "parameterDescription": "Implement or tool used for tillage",
  "unit": "code",
  "value": "TI012"
  "codeValueDescription": "Cultivator, row (TI012)",
}
```

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+          "code": "IROP",
+          "name": "irrigation_operation",
+          "value": "IR012",
+          "codeValueDescription": "Subsurface (burried) drip, mm"
```
`"unit": "code"`

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+        {
+          "code": "HACOM",
+          "name": "harvest_operat_component",
+          "value": "aboveground biomass"
```
ICASA says HACOM should be `"unit": "code"` but "aboveground biomass" is not one of the acceptable values. perhaps `"unit": "text"` for now until this new value can be added to ICASA

dlebauer replied:
going with

```
        {
          "code": "HACOM",
          "name": "harvest_operat_component",
          "unit": "code",
          "value": "canopy",
          "codeValueDescription": "entire aboveground biomass harvested, uncertain if 'canopy' is correct code"
        }
```

for now. Its not clear if 'canopy' has a more precise definition. But will follow up w/ ICASA folk.

BrapiCoordinatorSelby commented on data/events.json:
```diff
@@ -0,0 +1,492 @@
+    "eventParameters": [
+        {
+          "code": "S",
+          "name": "Dry Seed"
```
```
        {
          "code": "PLMA",
          "name": "planting_method",
          "value": "S",
          "unit": "code",
          "codeValueDescription": "Dry Seed"
        },
```

dlebauer committed:
Update data/events.json

Co-authored-by: Kristina Riemer <kristina.riemer@weecology.org>

dlebauer committed:
updates based on feedback from Peter Selby

dlebauer committed:
Merge branch 'events' of github.com:terraref/brapi into events

KristinaRiemer commented on data/events.json:
```diff
@@ -0,0 +1,558 @@
+    },
+    "eventParameters": [
+        {
+          "code": "HACOM",
```
Is this harvest event supposed to be of the same format as the very last one, starting on line 549? The code and name are flipped, and the unit is different.

dlebauer replied:
fixed, thanks for finding that!

KristinaRiemer commented on data/events.json:
```diff
@@ -0,0 +1,558 @@
+          "value": "Chlortaniliprole"
+        },
+        {
+          "code": "CH",
```
I can't find this code in the ICASA data dictionary. There's a *Chemicals application amount* (CHAMT)?

dlebauer replied:
In the Management_info tab? 

![image](https://user-images.githubusercontent.com/464871/171966515-204854a8-17fe-415d-b4c5-f51b9cdfddc5.png)

KristinaRiemer replied:
The code used here should actually be CHAMT. The CH code is an outdated linkage variable that shouldn't be used anymore (and hence doesn't show up in the Vmapper variable search).

KristinaRiemer commented on data/events.json:
```diff
@@ -0,0 +1,558 @@
+          "codeValueDescription": "Sorghum bicolor"
+        },
+        {
+          "code": "PLSL",
```
I'm not seeing this code in the ICASA data dictionary?

KristinaRiemer replied:
Or PLAL, PLNR, and PLSD, which are listed just below this entry.

dlebauer replied:
I think I have fixed these. Hopefully it isn't something that the AI code assistants (GitHub Copilot) injected!!!

dlebauer committed:
updated ICASA codes in events

dlebauer committed:
replaec deprecated CH with CHAMT

KristinaRiemer commented on data/events.json:
```diff
@@ -0,0 +1,562 @@
+        {
+          "code": "CHAMT",
+          "name": "chem_applic_level",
+          "unit": "l/ha",
```
The units for this ICASA variable are kg/ha, so this value could be converted using the chemical density.

dlebauer committed:
fixed Flubidamide application amount

dlebauer committed:
updated coragen application rate

KristinaRiemer committed:
Update Flubendiamide CHAMT unit

dlebauer commented:
@KristinaRiemer per brapi call today:

- we will use Code_Query rather than Code_Display because Code_Query excludes non-standard characters, and as name implies is intended for queries rather than display. This only affects the value PLTR# which has been changed back to PLTRno
- rename keys 
  - codeValueDescription --> valueDescription
  - parameterDescription --> description

dlebauer committed:
update events.json
- codeValueDescription --> valueDescription
- parameterDescription --> description

dlebauer committed:
Merge branch 'events' of github.com:terraref/brapi into events

dlebauer commented:
Ready to merge 

Thanks @KristinaRiemer and @BrapiCoordinatorSelby for your careful review, comments, and feedback.
